### ISSUE TITLE
Status 403 forbidden error

### ISSUE THREAD

**Issue by Su13n** [hareku/fanbox-dl#59]
I have tried using my FANBOXSESSID and manually setting the cookies. Started with Firefox, and tried Chrome when that didn't work. Still, for some reason I was unable to get authenticated even with the correct session ID. 
Lastly, I manually set the session ID cookie in Chrome which, too, resulted in a 403 forbidden error:

`https://www.fanbox.cc' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`.

**Comment by hareku** [comment]
Can you tell me how you are setting the flags for the fanbox-dl?

**Comment by BoiLudens** [comment]
Id been getting this as well, im not the OP but, 
```fanbox-dl --cookie "FANBOXSESSID=8675309_ABCDEFGHIJKLMNOP" --save-dir /home/user/Downloads --creator creatorNameHere --all```
was one i tried
the other was 
```fanbox-dl --sessid=8675309_ABCDEFGHIJKLMNOP --save-dir /home/user/Downloads --creator creatorNameHere --all```
and didnt work

```fanbox-dl  --save-dir /home/user/Downloads --creator creatorNameHere --all```
does work

And I was trying postman, with the cookie but unfortunately this is all i got
![image](https://github.com/user-attachments/assets/d62c604c-c5d8-4b20-b11d-195eb9b10653)

**Comment by SakkiWakki** [comment]
You need your cf_clearance value and your user agent as well

**Comment by TheSilverNuke** [comment]
I seem to be running into the same problem. I'm using [Version 0.27.3] [Windows_x86_64] in Powershell. Not all files are being downloaded. Some creators will end immediately, and others end after a page or two. The specific command I am using is as follows:

`.\fanbox-dl --cookie ***** --creator *****`

Now, I've attempted to expand on the command, but any lines with the verb "supporting" receive the following error:

`time=2025-03-03T19:22:46.795-07:00 level=INFO msg="Launching Pixiv FANBOX Downloader!" version=0.27.3 commit=46629ffeeaba56895bf6b9c252f4bea4e2988e24 date=2025-02-12T03:56:42Z
time=2025-03-03T19:22:47.049-07:00 level=ERROR msg="fanbox-dl Error" error="resolve creator IDs: list all creator IDs: list supporintg plans: status is 400 Bad Request"
time=2025-03-03T19:22:47.049-07:00 level=ERROR msg="The error log seems a bug, please open an issue on GitHub" url=https://github.com/hareku/fanbox-dl/issues`

The example provided in the description of the tool doesn't work for this exact reason. I'm not certain if the files are being skipped due to an incomplete command or if there is some other, deeper issue, with the tool. I've been using `--cookie` because `--sessid` doesn't work for me as it pops the 403 error specifically when I use that too. For clarity, the command listed in the description that I am trying to use is  thus:

`.\fanbox-dl --sessid xxxxx --save-dir ./content --creator creatornamehere --all`

This is the error I get with that command:

`time=2025-03-03T19:32:53.745-07:00 level=INFO msg="Launching Pixiv FANBOX Downloader!" version=0.27.3 commit=46629ffeeaba56895bf6b9c252f4bea4e2988e24 date=2025-02-12T03:56:42Z
time=2025-03-03T19:32:53.745-07:00 level=INFO msg="Start downloading" creator_id=**>insertcreatorid<**
time=2025-03-03T19:32:54.407-07:00 level=ERROR msg="fanbox-dl Error" error="failed downloading of \"**>insertcreatorid<**\": get pagination: status is 403 Forbidden"
time=2025-03-03T19:32:54.407-07:00 level=ERROR msg="The error log seems a bug, please open an issue on GitHub" url=https://github.com/hareku/fanbox-dl/issues`

**Comment by Ecollections** [comment]
There is another thread that has a great example command and some more clarity. [fanbox-dl example commane](https://github.com/hareku/fanbox-dl/issues/51#issuecomment-2446008375) I just recently reread the thread while troubleshooting an issue I was having.

I usually go to the home page of fanbox (not with the supported flag set) and `right click somewhere on the page > Inspect Element > select the application tab > storage folder > then the cookies folder` you should see a file in that folder with the fanbox url where the values are listed. A simpler way may be to install a "cookies downloader" extension from your browser's extensions/addons shop. That should give you a text file with the values you need.

**Comment by TheSilverNuke** [comment]
I'm still getting a 403 error when doing that. It seems to be doing okay up until I add --creator "creator_name" at the end of the string. Before that it just gives me an error stating that I need to resolve the creator ID, since the command example doesn't state which creator to download from.

**Comment by theshillito** [comment]
Until today, I just needed to connect to a VPN in Japan and Fanbox didn't use Cloudflare protection, so fanbox-dl worked. Today, however, even with the VPN enabled, I had to construct the cookie string as per the post referenced by Ecollections and add the user agent. The following cookies were needed (I put them in this order as key=value pairs separated by semicolons):

1. FANBOXSESSID
2. __cf_bm
3. cf_clearance
4. p_ab_d_id
5. p_ab_id
6. p_ab_id_2
7. privacy_policy_agreement
8. privacy_policy_notification

I assume the cloudflare stuff changes frequently so I can't have this automated in a scheduled task, but it might be possible to take some code from something like youtube-dl where the cookies are taken directly from the cookie store of a provided browser (Firefox, in my case).

**Comment by TheSilverNuke** [comment]
Yeah I don't know what it could be then. I'd very much like to download all of these files without going in one by one but I don't know what I'm doing wrong. I've made sure I was recently logged in, put in all 8 values listed, my user agent, the creator, and ensured I was connected to a VPN with a Japan end point, up to date on the program. I'm sure it will turn out to be something incredibly simple that I overlooked but I can't think of what might be causing this problem.

**Comment by Tsoomer** [comment]
I'm having this issue as well, despite setting all the cookies as described here and setting the user agent to my browser's user agent

**Comment by Whatintheworldbro** [comment]
C:\Users\myusername\Downloads>fanbox-dl --sessid 103108777_SAGQRY9fxZwTe6EWuoscsb4dLkI9q21O --save-dir \Users\myusername\downloads  --creator creatorname --all
time=2025-05-19T17:00:55.157-07:00 level=INFO msg="Launching Pixiv FANBOX Downloader!" version=0.27.3 commit=46629ffeeaba56895bf6b9c252f4bea4e2988e24 date=2025-02-12T03:56:42Z
time=2025-05-19T17:00:55.158-07:00 level=INFO msg="Start downloading" creator_id=creatorname
time=2025-05-19T17:00:55.297-07:00 level=ERROR msg="fanbox-dl Error" error="failed downloading of \"creatorname\": get pagination: status is 403 Forbidden"
time=2025-05-19T17:00:55.297-07:00 level=ERROR msg="The error log seems a bug, please open an issue on GitHub" url=https://github.com/hareku/fanbox-dl/issues

i need some help as well, keeps giving 403

**Comment by xzario** [comment]
I encountered an issue which says error: status code 400. It was random, either it was 400, 401, or 403. Curious, I tried to download from other artists, and fanbox-dl unexpectedly ran normally. I found the root cause of the problem (in my case) was a problematic post in a specific artist which somehow causes an error to fanbox-dl and halts its operation.

I added --skip-on-error in the command, and now it works like a charm. It doesn't download the problematic post but downloads the rest of the artist's content.

Here's the problematic post that causes fanbox-dl to stop somehow. I've never seen a case like this with other posts.
https://[downloads.fanbox.cc/images/post/7995101/DCrxu1tzcxK7J24rjsKYAUIB.jpeg](https://downloads.fanbox.cc/images/post/7995101/DCrxu1tzcxK7J24rjsKYAUIB.jpeg)

**Comment by RogRog00** [comment]
Well, it was working most of the time when i used --sessid and --user-agent, but since I would say yesterday, I only encounter the ""fanbox-dl Error" error="failed downloading of \"xxxx\": handle page: handle post: get post: status is 403 Forbidden" error. I've checked my user-agent, I've re-asked for an ID by logging out and in, and tried without the user-agent (because why not). And all of this without any success.
